tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -633,11 +633,17 @@ static int bits_logical(void)
 
 /* gpio_pin_get_config()
  */
+#ifdef CONFIG_GPIO_GET_CONFIG
 static int pin_get_config(void)
 {
 	gpio_flags_t flags_get = 0;
 	gpio_flags_t flags_set;
 	int rc;
+	gpio_flags_t flags_mask = GPIO_INT_MASK | GPIO_OUTPUT_INIT_LOGICAL |
+				  GPIO_OUTPUT_INIT_HIGH | GPIO_OUTPUT_INIT_LOW |
+				  GPIO_OUTPUT | GPIO_INPUT |
+				  GPIO_PULL_UP | GPIO_PULL_DOWN |
+				  GPIO_SINGLE_ENDED | GPIO_LINE_OPEN_DRAIN | GPIO_ACTIVE_LOW;
```

Comment:
So the idea is to mask out SoC-specific GPIO flags? Why not just use `GENMASK` to generate a bit mask excluding those? The above would need to be updated every time we add a new flag?

```
/**
 * @brief Provides a type to hold GPIO devicetree flags.
 *
 * All GPIO flags that can be expressed in devicetree fit in the low 16
 * bits of the full flags field, so use a reduced-size type to record
 * that part of a GPIOS property.
 *
 * The lower 8 bits are used for standard flags. The upper 8 bits are reserved
 * for SoC specific flags.
 */
typedef uint16_t gpio_dt_flags_t;

/**
 * @brief Provides a type to hold GPIO configuration flags.
 *
 * This type is sufficient to hold all flags used to control GPIO
 * configuration, whether pin or interrupt.
 */
typedef uint32_t gpio_flags_t;

```

Reply:
> So the idea is to mask out SoC-specific GPIO flags?

In this case the problem was with voltage flags from device-tree. If you don't specify any voltage (so flag is GPIO_VOLTAGE_DEFAULT), you can read a real voltage from board registers (GPIO_VOLTAGE_3P3).

Reply:
OK, use GENMASK